### ISSUE TITLE
KeyError: 'ner_crf' in evaluate.py during crossvalidation 

### ISSUE THREAD

**Issue by TatianaParshina** [RasaHQ/rasa_nlu#1603]
Hello,

**Operating system** (windows, osx, ...): windows 10

**Content of model configuration file**:
```yml
language: "en"

pipeline:
- name: "tokenizer_whitespace"
- name: "ner_crf"
- name: "intent_featurizer_count_vectors"
- name: "intent_classifier_tensorflow_embedding"
intent_tokenization_flag: true
intent_split_symbol: "_"

```

**Issue**:
I am trying to evaluate model by running command:
```
python -m rasa_nlu.evaluate --data data/nlu_data.json --config nlu_config.yml --mode crossvalidation --folds 5
```
However, I got issue:
```
Traceback (most recent call last):
 File "C:\ProgramData\Anaconda3\lib\runpy.py", line 193, in _run_module_as_main
 "__main__", mod_spec)
 File "C:\ProgramData\Anaconda3\lib\runpy.py", line 85, in _run_code
 exec(code, run_globals)
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 857,
in <module>
 main()
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 834,
in main
 data, int(cmdline_args.folds), nlu_config)
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 720,
in run_cv_evaluation
 interpreter, train)
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 684,
in combine_entity_result
 current_result = compute_entity_metrics(interpreter, data)
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 777,
in compute_entity_metrics
 merged_predictions = merge_labels(aligned_predictions, extractor)
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 307,
in merge_labels
 for ap in aligned_predictions]
 File "C:\ProgramData\Anaconda3\lib\site-packages\rasa_nlu\evaluate.py", line 307,
in <listcomp>
 for ap in aligned_predictions]
KeyError: 'ner_crf'
```

**Comment by wochinge** [comment]
* Which rasa_nlu version are you using? 
* did you install `sklearn-crfsuite`?

**Comment by amaurymercier** [comment]
I got the exact same issue. It happens with (out of the box) tensorflow embedding pipeline and crossvalidation.

packages :
- rasa-nlu : 0.13.8
- scikit-learn : 0.19.0
- sklearn-crfsuite : 0.3.6

**Comment by TatianaParshina** [comment]
Used packages:
- rasa_nlu 0.13.8
- sklearn_crfsuite 0.3.6

**Comment by wochinge** [comment]
@amaurymercier Are you also running Windows or a different OS?

Could you please try to open your python console and do `import sklearn_crfsuite`. Do you get any error there?

**Comment by amaurymercier** [comment]
I'm on Ubuntu 18.04 with python 3.6.
`import sklearn_crfsuite` works fine :)

**Comment by wochinge** [comment]
And what was the command you used to install rasa nlu? `pip install rasa_nlu` or something like `pip install rasa_nlu[spacy]`?

**Comment by amaurymercier** [comment]
only `pip install rasa_nlu` if I remember well, but I did install spacy & spacy models afterwards.

**Comment by wochinge** [comment]
Mhm, I cannot reproduce this on my machine. Can you please run it with the debug flag and post the output here? `python -m rasa_nlu.evaluate --data data/nlu.json --config nlu_config.yml --mode crossvalidation --folds 1 --debug` (folds == 1 should be enough I guess)

**Comment by amaurymercier** [comment]
I put `--folds 2` as it was mandatory.

I got more meaningful error messages :

`
/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/sklearn/metrics/classification.py:1137: UndefinedMetricWarning: Recall and F-score are ill-defined and being set to 0.0 in labels with no true samples.
  'recall', 'true', average, warn_for)
/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/sklearn/metrics/classification.py:1137: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true samples.
  'recall', 'true', average, warn_for)
2019-01-03 12:10:42 DEBUG    rasa_nlu.extractors.crf_entity_extractor  - Inconsistent BILOU tagging found, B- tag, L- tag pair encloses multiple entity classes.i.e. [B-a, I-b, L-a] instead of [B-a, I-a, L-a].
Assuming B- class is correct.
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 857, in <module>
    main()
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 834, in main
    data, int(cmdline_args.folds), nlu_config)
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 720, in run_cv_evaluation
    interpreter, train)
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 684, in combine_entity_result
    current_result = compute_entity_metrics(interpreter, data)
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 777, in compute_entity_metrics
    merged_predictions = merge_labels(aligned_predictions, extractor)
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 307, in merge_labels
    for ap in aligned_predictions]
  File "/home/amaury/.virtualenvs/pychat/lib/python3.6/site-packages/rasa_nlu/evaluate.py", line 307, in <listcomp>
    for ap in aligned_predictions]
KeyError: 'ner_crf'
`

**Comment by wochinge** [comment]
Thanks. How much training examples do you have per intent and how much of them have annotations? Can you maybe share some (minimal) nlu training data?

**Comment by amaurymercier** [comment]
I think I got it.

The error happens when there is an utterance without any letters (only spaces and punctuation, for instance : i had the utterance "? ??"). When I removed it, it went fine.

**Comment by imLew** [comment]
Resolved in PR#1689
